### ISSUE TITLE
Improve markdown component's prop editor 

### ISSUE THREAD

**Issue by prakhargupta1** [mui/mui-toolpad#1310]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

Currently, the prop editor is a single line text box. Most of the time the content will be multiline. This issue is about how to handle this experience. Some options:

1. We allow expanding the side panel and make the editor big.
2. We create an indicator that expands the text box to a bigger dialog box (like retool)

Note: As this experience will be introduced to other components in the future, we should aim to keep it standard.

**Comment by bytasv** [comment]
Just a quick note, we already allow drag-resizing right panel:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/437214/200776201-ab2d4f54-8e2a-4018-9a80-82eaff60d2e8.png">

**Comment by prakhargupta1** [comment]
Thanks for highlighting it. It would great to prominently show that the panel expands.
Additionally, I like how this UI on GitHub textboxes work. We can use it as well. So, the user can expand the panel, and can also expand the text box.

<img width="45" alt="Screenshot 2022-11-09 at 2 31 33 PM" src="https://user-images.githubusercontent.com/92228082/200786184-3b86b35d-ee58-4d92-bcb5-011cdbe2137e.png">

**Comment by Janpot** [comment]
I don't believe requiring users to reorganize the UI in two dimensions just to be able to edit markdown somewhat more ergonomically is a great UX. I propose we stick with a modal for now which will give us with one click a predictably sized surface to build a full featured markdown editor into.
